### ISSUE TITLE
"invalid input constraint '0' in asm" in arch/riscv/include/asm/errata_list.h

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1641]
After commit [a35707c3d850](https://git.kernel.org/riscv/c/a35707c3d850dda0ceefb75b1b3bd191921d5765) ("riscv: add memory-type errata for T-Head") in -next:

```
$ make -skj"$(nproc)" ARCH=riscv LLVM=1 allmodconfig all
In file included from arch/riscv/kernel/asm-offsets.c:10:
In file included from ./include/linux/mm.h:29:
In file included from ./include/linux/pgtable.h:6:
In file included from ./arch/riscv/include/asm/pgtable.h:114:
./arch/riscv/include/asm/pgtable-64.h:210:2: error: invalid input constraint '0' in asm
        ALT_THEAD_PMA(prot_val);
        ^
./arch/riscv/include/asm/errata_list.h:88:4: note: expanded from macro 'ALT_THEAD_PMA'
        : "0"(_val),                                                    \
          ^
In file included from arch/riscv/kernel/asm-offsets.c:10:
In file included from ./include/linux/mm.h:29:
In file included from ./include/linux/pgtable.h:6:
./arch/riscv/include/asm/pgtable.h:257:2: error: invalid input constraint '0' in asm
        ALT_THEAD_PMA(prot_val);
        ^
./arch/riscv/include/asm/errata_list.h:88:4: note: expanded from macro 'ALT_THEAD_PMA'
        : "0"(_val),                                                    \
          ^
In file included from arch/riscv/kernel/asm-offsets.c:10:
In file included from ./include/linux/mm.h:29:
In file included from ./include/linux/pgtable.h:6:
./arch/riscv/include/asm/pgtable.h:300:2: error: invalid input constraint '0' in asm
        ALT_THEAD_PMA(prot_val);
        ^
./arch/riscv/include/asm/errata_list.h:88:4: note: expanded from macro 'ALT_THEAD_PMA'
        : "0"(_val),                                                    \
          ^
In file included from arch/riscv/kernel/asm-offsets.c:10:
In file included from ./include/linux/mm.h:29:
In file included from ./include/linux/pgtable.h:6:
./arch/riscv/include/asm/pgtable.h:418:2: error: invalid input constraint '0' in asm
        ALT_THEAD_PMA(newprot_val);
        ^
./arch/riscv/include/asm/errata_list.h:88:4: note: expanded from macro 'ALT_THEAD_PMA'
        : "0"(_val),                                                    \
          ^
...
```

**Comment by nickdesaulniers** [comment]
See upstream response.

**Comment by nickdesaulniers** [comment]
Was there a kernel ML thread about this?

**Comment by nathanchance** [comment]
> Was there a kernel ML thread about this?

A thread for the original patch or this issue? I didn't report anything upstream, I just filed it here initially. The patch thread is here: https://lore.kernel.org/20220511192921.2223629-13-heiko@sntech.de/

I'm going to be offline for the rest of today but I can either send Jessica's suggested diff (if it works) or report the problem upstream tomorrow, unless someone else beats me to it today.

**Comment by nathanchance** [comment]
We still need to resolve this build failure so I am reopening the issue, even if there is nothing to fix on the toolchain side.

**Comment by nathanchance** [comment]
https://lore.kernel.org/20220518184529.454008-1-nathan@kernel.org/

**Comment by nathanchance** [comment]
Palmer has picked this up: https://git.kernel.org/riscv/c/e83031564137cf37e07c2d10ad468046ff48a0cf

**Comment by nathanchance** [comment]
This landed in 5.19-rc4: https://git.kernel.org/linus/e83031564137cf37e07c2d10ad468046ff48a0cf
